File: src/main/java/com/jigumulmi/admin/place/dto/response/AdminPlaceListResponseDto.java
package com.jigumulmi.admin.place.dto.response;

import com.jigumulmi.common.PageDto;
import com.jigumulmi.place.domain.Place;
import com.jigumulmi.place.dto.response.PlaceCategoryDto;
import com.jigumulmi.place.dto.response.PlaceResponseDto;
import com.jigumulmi.place.dto.response.SubwayStationResponseDto;
import java.util.List;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Getter
@Builder
public class AdminPlaceListResponseDto {

    @Getter
    @SuperBuilder
    @NoArgsConstructor
    public static class PlaceDto extends PlaceResponseDto {

        private Boolean isApproved;

        public static PlaceDto from(Place place) {
            SubwayStationResponseDto subwayStationResponseDto;
            if (place.getSubwayStationPlaceList().isEmpty()) {
                subwayStationResponseDto = null;
            } else {
                subwayStationResponseDto = SubwayStationResponseDto.fromMainStation(
                    place.getSubwayStationPlaceList().getFirst().getSubwayStation()
                );
            }

            return PlaceDto.builder()
                .id(place.getId())
                .name(place.getName())
                .position(
                    PositionDto.builder().latitude(place.getLatitude())
                        .longitude(place.getLongitude())
                        .build()
                )
                .subwayStation(subwayStationResponseDto)
                .categoryList(place.getCategoryMappingList().stream()
                    .map(PlaceCategoryDto::fromPlaceCategoryMapping).toList())
                .isApproved(place.getIsApproved())
                .build();
        }
    }

    private PageDto page;
    private List<PlaceDto> data;
}

File: admin/src/main/kotlin/com/jigumulmi/place/dto/response/AdminPlaceListResponseDto.kt
package com.jigumulmi.place.dto.response

import com.jigumulmi.common.AdminPagedResponseDto
import com.jigumulmi.place.domain.Place
import com.jigumulmi.place.dto.ImageDto
import com.jigumulmi.place.dto.PositionDto
import com.jigumulmi.place.dto.response.AdminPlaceListResponseDto.PlaceDto

data class AdminPlaceListResponseDto(
    override val page: PageDto,
    override val data: List<PlaceDto>
) : AdminPagedResponseDto<PlaceDto>(page, data) {
    data class PlaceDto(
        val id: Long,
        val name: String? = null,
        val imageList: List<ImageDto> = emptyList(),
        val position: PositionDto = PositionDto(),
        val subwayStation: SubwayStationResponseDto? = null,
        val categoryList: List<PlaceCategoryDto> = emptyList(),
        val isApproved: Boolean
    ) {
        companion object {
            fun from(place: Place): PlaceDto {
                val subwayStationResponseDto = if (place.subwayStationPlaceList.isEmpty()) {
                    null
                } else {
                    SubwayStationResponseDto.fromMainStation(
                        place.subwayStationPlaceList.first().subwayStation
                    )
                }

                return PlaceDto(
                    id = place.id,
                    name = place.name,
                    position = PositionDto.builder()
                        .latitude(place.latitude)
                        .longitude(place.longitude)
                        .build(),
                    subwayStation = subwayStationResponseDto,
                    categoryList = place.categoryMappingList.map(PlaceCategoryDto::fromPlaceCategoryMapping),
                    isApproved = place.isApproved
                )
            }
        }
    }
}
